extern/detray/CMakeLists.txt

```diff
@@ -0,0 +1,41 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# CMake include(s).
+cmake_minimum_required( VERSION 3.14 )
+include( FetchContent )
+
+# Tell the user what's happening.
+message( STATUS "Building Detray as part of the TRACCC project" )
+
+# Declare where to get Detray from.
+set( TRACCC_DETRAY_SOURCE
+   "GIT_REPOSITORY;https://github.com/acts-project/detray.git;GIT_TAG;3aafe70"
```

Comment:
Any reason why we fetch this via Git instead of downloading it as a tarball?

Reply:
Again, explanation in the description of the PR. :stuck_out_tongue: I'll update it shortly...

Reply:
You can download a tarball for any revision in Github, that was my main confusion. :D